content/client-lib-development-guide/features.textile

```diff
@@ -74,21 +74,32 @@ h3(#restclient). RestClient
 ** @(RSC15b)@ This fallback behaviour only applies to the default @rest.ably.io@ endpoint; if either @host@ or @environment@ options are set, no fallback host is used.
 ** @(RSC15c)@ See the "default fallback hosts":https://github.com/ably/ably-java/blob/master/src/io/ably/transport/Defaults.java#L7 and the "fallback mechanism implemented in Java for POST requests":https://github.com/ably/ably-java/blob/master/src/io/ably/http/Http.java#L204-L224 as an example.
 ** @(RSC15d)@ Automatic retries to fallback hosts will not occur for any other HTTP status codes. For example, if the credentials are invalid or the maxmimum message length has been exceeded, then resending the request to an alternative data center will not fix the problem and will only increase the load on other data centers. See the "complete list of Ably error codes":https://github.com/ably/ably-common/blob/master/protocol/errors.json#L11 and a "discussion in the Java library about this":https://github.com/ably/ably-java/issues/14
+* @(RSC17)@ When instancing the library, if a @clientId@ attribute is set in @ClientOptions@, then the @Auth#clientId@ attribute will contain the provided @clientId@
 
 h3(#rest-auth). Auth
 
 * @(RSA1)@ Basic Auth connects over HTTPS by default.  If HTTP and Basic Auth are used, an exception will be raised
-* @(RSA11)@ When using Basic Auth, the API key is sent as the username & password portion of an HTTP URL or in the @Authorization: Basic@ header with a Base64 encoded value
+* @(RSA11)@ When using Basic Auth, the API key is sent as the username & password portion of an HTTPS URL or in the @Authorization: Basic@ header with a Base64 encoded value
 * @(RSA2)@ Basic Auth is the default authentication scheme if an API key exists
 * @(RSA3)@ Token Auth:
 ** @(RSA3a)@ Can be used over HTTP or HTTPs
 ** @(RSA3b)@ For REST requests, the token string is Base64 encoded and used in the @Authorization: Bearer@ header
 ** @(RSA3c)@ For Realtime connections, the querystring param @access_token@ is appended to the URL endpoint
 * @(RSA4)@ Token Auth is the default if option @useTokenAuth@ is set to true, a @clientId@ is specified, @authUrl@ or @authCallback@ is configured, or a an explicit @token@ is provided
-* @(RSA14)@ If Token Auth is the default authentication, an exception will be raised if a token is not provided or there is no means to generate a token. For example, if only the option @useTokenAuth@ is specified, and thus a @key@ is not provided, then the client library is unable to authenticate or issue a token
+* @(RSA14)@ If Token Auth is the default authentication scheme, an exception will be raised if a token is not provided or there is no means to generate a token. For example, if only the option @useTokenAuth@ is specified, and a @key@ is not provided, then the client library is unable to authenticate or issue a token
```

Comment:
Again the use of "default" here is confused.

If Token Auth is selected, either because it was explicitly specified in @useTokenAuth@, or was selected by default (eg by clientId being specified), then an exception will be raised ....